packages/roosterjs-editor-types/lib/interface/IEditor.ts

```diff
@@ -499,6 +499,13 @@ export default interface IEditor {
      */
     getStyleBasedFormatState(node?: Node): StyleBasedFormatState;
 
+    /**
+     * Get the pendable format such as underline and bold
+     * @param forceGetStateFromDOM If set to true, will force get the format state from DOM tree.
+     * @return The pending format state of editor.
+     */
+    getPendableFormatState(forceGetStateFromDOM: boolean): PendableFormatState;
```

Comment:
forceGetStateFromDOM should still be optional for IEditor and Editor. We just need to make it required for core API.

The principle here is, for common public API such as Editor API, format API, we should make it easy to use. For core API which is more "internally used", we want to make it strict and explicit.